rpc/proto/persistence.proto

```diff
@@ -0,0 +1,36 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "io.openebs.mayastor";
+option java_outer_classname = "PersistenceProto";
+
+package persistence;
+
+enum ChildState {
+  Init = 0;
+  ConfigInvalid = 1;
+  Open = 2;
+  Destroying = 3;
+  Closed= 4;
+  Faulted = 5;
+}
+
+enum Reason {
+  Unknown = 0;
+  OutOfSync = 1;
+  CantOpen = 2;
+  RebuildFailed = 3;
+  IoError = 4;
+  Rpc = 5;
+}
+
+message ChildInfo {
+  string uuid = 1;
+  ChildState state = 2;
+  Reason reason = 3;
+}
```

Comment:
It's not a problem but I'm wondering if we shouldn't keep the api minimal in a sense that we should avoid everything that is strictly not necessary. AFAIK all we need is:
* clean shutdown flag
* uuid of replica
* state of child and in particular just two states (Open and not open)

Adding new things to grpc api is easy and backward compatible but old things that are not used will become a maintenance burden.

Reply:
We tried to keep the states close to what mayastor has to gives us a bit more observability of what "went wrong" when a child is not healthy anymore. For example, in the future we might be able to do partials rebuilds so knowing that the child was not faulted but only out of sync would be useful.
Anyway, MOAC is only reading these fields from mayastor so I'll add @paulyoong who can make decisions about the data format.

Reply:
I think we should keep it. The new control plane is likely to be more opinionated on things in general and so will probably require more detailed information to be passed up from Mayastor. That future work may require the gRPC layer to be overhauled anyway, which I guess we **could** introduce as a breaking change, therefore the backwards compatibility issue wouldn't arise.

Reply:
👍